From: Sean Wang <sean.wang@mediatek.com>
Date: Thu, 16 Dec 2021 05:25:37 +0800
Subject: [PATCH] mt76: mt7921s: fix suspend error with enlarging mcu timeout
 value
References: bsc#1012628
Patch-mainline: 5.16.3
Git-commit: 1bb42a354d8ca2888c7c2fcbf0add410176a33dc

[ Upstream commit 1bb42a354d8ca2888c7c2fcbf0add410176a33dc ]

Fix the false positive suspend error that may occur on mt7921s
with enlarging mcu timeout value.

The reason why we have to enlarge mcu timeout from HZ / 3 to HZ is
we should consider the additional overhead caused by running
concurrently with btmtksdio (a MT7921 bluetooth SDIO driver) that
would compete for the same SDIO bus in process context to complete
the suspend procedure.

Fixes: 48fab5bbef40 ("mt76: mt7921: introduce mt7921s support")
Signed-off-by: Sean Wang <sean.wang@mediatek.com>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/net/wireless/mediatek/mt76/mt7921/mcu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/mediatek/mt76/mt7921/mcu.c b/drivers/net/wireless/mediatek/mt76/mt7921/mcu.c
index e741db152c0d..1cc1c32ca258 100644
--- a/drivers/net/wireless/mediatek/mt76/mt7921/mcu.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7921/mcu.c
@@ -229,7 +229,7 @@ int mt7921_mcu_fill_message(struct mt76_dev *mdev, struct sk_buff *skb,
 	if (cmd == MCU_UNI_CMD(HIF_CTRL) ||
 	    cmd == MCU_UNI_CMD(SUSPEND) ||
 	    cmd == MCU_UNI_CMD(OFFLOAD))
-		mdev->mcu.timeout = HZ / 3;
+		mdev->mcu.timeout = HZ;
 	else
 		mdev->mcu.timeout = 3 * HZ;
 
-- 
2.34.1

